api/serializers/__init__.py

```diff
@@ -105,7 +105,39 @@ class Meta:
         ]
 
 
-class ProfileSerializer(serializers.ModelSerializer):
+class StrictReadOnlyFieldsMixin:
+    """Raises a validation error (400) if read only fields are in the body of PUT/PATCH requests.
+
+    This class comes from https://github.com/encode/django-rest-framework/issues/1655#issuecomment-1197033853,
+    where different solutions to mitigating 200 response codes in read-only fields are discussed.
+    """
+
+    def validate(self, attrs):
```

Comment:
(suggestion - blocking) Can you add tests for this code?
"happy" case - where the attribute that is read only on the model and the attribute that is read only on the serializer is passed to the `validate()` and returns the error
happy case - where the attribute that is editable is passed to the `validate()` and DOES NOT returns the error

Reply:
Added these cases (https://github.com/mozilla/fx-private-relay/pull/3795#discussion_r1317838049, https://github.com/mozilla/fx-private-relay/pull/3795#discussion_r1317839215) and some additional ones.